[Bug]: Failed to persist optimizing process records

### What happened?

Failure to persist the optimizating process record to the database, even though the optimization results have been committed
![image](https://github.com/NetEase/amoro/assets/54210725/9bb993d5-2bc7-44d2-8774-4cd6961b2c7f)
![image](https://github.com/NetEase/amoro/assets/54210725/97ef535f-c12f-4bcb-b72a-e09f00aa7b39)
![image](https://github.com/NetEase/amoro/assets/54210725/b4b34edf-6b69-4601-ad7d-c70b0f503ace)


### Affects Versions

master/0.5.x

### What engines are you seeing the problem on?

AMS

### How to reproduce

_No response_

### Relevant log output

```shell
### Error updating database.  Cause: java.sql.SQLNonTransientConnectionException: No operations allowed after connection closed.
### The error may exist in com/netease/arctic/server/persistence/mapper/TableMetaMapper.java (best guess)
### The error may involve com.netease.arctic.server.persistence.mapper.TableMetaMapper.updateTableRuntime
### The error occurred while executing an update
### SQL: UPDATE table_runtime SET current_snapshot_id = ?, current_change_snapshotId =?, last_optimized_snapshotId = ?, last_optimized_change_snapshotId = ?, last_major_optimizing_time = ?, last_minor_optimizing_time = ?, last_full_optimizing_time = ?, optimizing_status = ?, optimizing_status_start_time = ?, optimizing_process_id = ?, optimizer_group = ?, table_config = ?, pending_input = ? WHERE table_id = ?
### Cause: java.sql.SQLNonTransientConnectionException: No operations allowed after connection closed.
        at com.netease.arctic.server.exception.ArcticRuntimeException.wrap(ArcticRuntimeException.java:123) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.persistence.PersistentBase.doAs(PersistentBase.java:52) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.TableRuntime.persistUpdatingRuntime(TableRuntime.java:299) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.TableRuntime.lambda$setPendingInput$6(TableRuntime.java:174) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.persistence.StatedPersistentBase.invokeConsisitency(StatedPersistentBase.java:31) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.TableRuntime.setPendingInput(TableRuntime.java:169) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.executor.TableRuntimeRefreshExecutor.tryEvaluatingPendingInput(TableRuntimeRefreshExecutor.java:64) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.executor.TableRuntimeRefreshExecutor.handleStatusChanged(TableRuntimeRefreshExecutor.java:53) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.lambda$fireStatusChanged$0(RuntimeHandlerChain.java:42) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.doSilently(RuntimeHandlerChain.java:84) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.fireStatusChanged(RuntimeHandlerChain.java:42) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.fireStatusChanged(RuntimeHandlerChain.java:44) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.fireStatusChanged(RuntimeHandlerChain.java:44) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.fireStatusChanged(RuntimeHandlerChain.java:44) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.fireStatusChanged(RuntimeHandlerChain.java:44) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.fireStatusChanged(RuntimeHandlerChain.java:44) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.fireStatusChanged(RuntimeHandlerChain.java:44) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.DefaultTableService.handleTableChanged(DefaultTableService.java:285) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.TableRuntime.lambda$completeProcess$11(TableRuntime.java:236) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.persistence.StatedPersistentBase.invokeConsisitency(StatedPersistentBase.java:31) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.TableRuntime.completeProcess(TableRuntime.java:220) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.optimizing.OptimizingQueue$TableOptimizingProcess.lambda$persistProcessCompleted$16(OptimizingQueue.java:591) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.persistence.PersistentBase.doAsTransaction(PersistentBase.java:61) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.optimizing.OptimizingQueue.access$800(OptimizingQueue.java:55) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.optimizing.OptimizingQueue$TableOptimizingProcess.persistProcessCompleted(OptimizingQueue.java:587) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.optimizing.OptimizingQueue$TableOptimizingProcess.commit(OptimizingQueue.java:523) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.executor.OptimizingCommitExecutor.execute(OptimizingCommitExecutor.java:32) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.executor.BaseTableExecutor.executeTask(BaseTableExecutor.java:53) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.executor.BaseTableExecutor.lambda$scheduleIfNecessary$4(BaseTableExecutor.java:65) ~[amoro-ams-server-0.5.0.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_332]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_332]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_332]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_332]
        at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_332]
Caused by: org.apache.ibatis.exceptions.PersistenceException: 
### Error updating database.  Cause: java.sql.SQLNonTransientConnectionException: No operations allowed after connection closed.
### The error may exist in com/netease/arctic/server/persistence/mapper/TableMetaMapper.java (best guess)
### The error may involve com.netease.arctic.server.persistence.mapper.TableMetaMapper.updateTableRuntime
### The error occurred while executing an update
### SQL: UPDATE table_runtime SET current_snapshot_id = ?, current_change_snapshotId =?, last_optimized_snapshotId = ?, last_optimized_change_snapshotId = ?, last_major_optimizing_time = ?, last_minor_optimizing_time = ?, last_full_optimizing_time = ?, optimizing_status = ?, optimizing_status_start_time = ?, optimizing_process_id = ?, optimizer_group = ?, table_config = ?, pending_input = ? WHERE table_id = ?
### Cause: java.sql.SQLNonTransientConnectionException: No operations allowed after connection closed.
        at org.apache.ibatis.exceptions.ExceptionFactory.wrapException(ExceptionFactory.java:30) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.session.defaults.DefaultSqlSession.update(DefaultSqlSession.java:199) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.binding.MapperMethod.execute(MapperMethod.java:67) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.binding.MapperProxy$PlainMethodInvoker.invoke(MapperProxy.java:152) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.binding.MapperProxy.invoke(MapperProxy.java:85) ~[mybatis-3.5.6.jar:3.5.6]
        at com.sun.proxy.$Proxy40.updateTableRuntime(Unknown Source) ~[?:?]
        at com.netease.arctic.server.table.TableRuntime.lambda$persistUpdatingRuntime$15(TableRuntime.java:299) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.persistence.PersistentBase.doAs(PersistentBase.java:48) ~[amoro-ams-server-0.5.0.jar:?]
        ... 34 more
--
2023-08-22 02:01:08,238 INFO [ASYNC-TableRuntimeRefreshExecutor-9] [com.netease.arctic.server.table.TableRuntime] [] - Refreshing table dls_iceberg.xxx.yyy(tableId=843744) with base snapshot id 5270774535475818926
2023-08-22 02:01:08,238 WARN [ASYNC-OptimizingCommitExecutor-2] [com.netease.arctic.server.optimizing.OptimizingQueue] [] - dls_iceberg.xxx.yyy(tableId=843744) Commit optimizing failed 
org.apache.ibatis.exceptions.PersistenceException: 
### Error rolling back transaction.  Cause: java.sql.SQLNonTransientConnectionException: No operations allowed after connection closed.
### Cause: java.sql.SQLNonTransientConnectionException: No operations allowed after connection closed.
        at org.apache.ibatis.exceptions.ExceptionFactory.wrapException(ExceptionFactory.java:30) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.session.defaults.DefaultSqlSession.rollback(DefaultSqlSession.java:243) ~[mybatis-3.5.6.jar:3.5.6]
        at com.netease.arctic.server.persistence.NestedSqlSession.rollback(NestedSqlSession.java:59) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.persistence.PersistentBase.doAsTransaction(PersistentBase.java:65) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.optimizing.OptimizingQueue.access$800(OptimizingQueue.java:55) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.optimizing.OptimizingQueue$TableOptimizingProcess.persistProcessCompleted(OptimizingQueue.java:587) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.optimizing.OptimizingQueue$TableOptimizingProcess.commit(OptimizingQueue.java:523) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.executor.OptimizingCommitExecutor.execute(OptimizingCommitExecutor.java:32) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.executor.BaseTableExecutor.executeTask(BaseTableExecutor.java:53) ~[amoro-ams-server-0.5.0.jar:?]
        at com.netease.arctic.server.table.executor.BaseTableExecutor.lambda$scheduleIfNecessary$4(BaseTableExecutor.java:65) ~[amoro-ams-server-0.5.0.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_332]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_332]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_332]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_332]
        at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_332]
Caused by: java.sql.SQLNonTransientConnectionException: No operations allowed after connection closed.
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:110) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:73) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:73) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.jdbc.ConnectionImpl.rollback(ConnectionImpl.java:1844) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at org.apache.commons.dbcp2.DelegatingConnection.rollback(DelegatingConnection.java:781) ~[commons-dbcp2-2.9.0.jar:2.9.0]
        at org.apache.commons.dbcp2.DelegatingConnection.rollback(DelegatingConnection.java:781) ~[commons-dbcp2-2.9.0.jar:2.9.0]
        at org.apache.ibatis.transaction.jdbc.JdbcTransaction.rollback(JdbcTransaction.java:82) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.executor.BaseExecutor.rollback(BaseExecutor.java:256) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.executor.CachingExecutor.rollback(CachingExecutor.java:126) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.session.defaults.DefaultSqlSession.rollback(DefaultSqlSession.java:240) ~[mybatis-3.5.6.jar:3.5.6]
        ... 15 more
Caused by: com.mysql.cj.exceptions.ConnectionIsClosedException: No operations allowed after connection closed.
        at sun.reflect.GeneratedConstructorAccessor103.newInstance(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_332]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_332]
        at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:61) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:105) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:151) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.NativeSession.checkClosed(NativeSession.java:761) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.jdbc.ConnectionImpl.checkClosed(ConnectionImpl.java:568) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at com.mysql.cj.jdbc.ConnectionImpl.rollback(ConnectionImpl.java:1798) ~[mysql-connector-java-8.0.30.jar:8.0.30]
        at org.apache.commons.dbcp2.DelegatingConnection.rollback(DelegatingConnection.java:781) ~[commons-dbcp2-2.9.0.jar:2.9.0]
        at org.apache.commons.dbcp2.DelegatingConnection.rollback(DelegatingConnection.java:781) ~[commons-dbcp2-2.9.0.jar:2.9.0]
        at org.apache.ibatis.transaction.jdbc.JdbcTransaction.rollback(JdbcTransaction.java:82) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.executor.BaseExecutor.rollback(BaseExecutor.java:256) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.executor.CachingExecutor.rollback(CachingExecutor.java:126) ~[mybatis-3.5.6.jar:3.5.6]
        at org.apache.ibatis.session.defaults.DefaultSqlSession.rollback(DefaultSqlSession.java:240) ~[mybatis-3.5.6.jar:3.5.6]
        ... 15 more
```


### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/ams/server/src/main/java/com/netease/arctic/server/table/executor/TableRuntimeRefreshExecutor.java
+++ b/ams/server/src/main/java/com/netease/arctic/server/table/executor/TableRuntimeRefreshExecutor.java
@@ -46,14 +46,6 @@ public class TableRuntimeRefreshExecutor extends BaseTableExecutor {
     return Math.min(tableRuntime.getOptimizingConfig().getMinorLeastInterval() * 4L / 5, interval);
   }
 
-  @Override
-  public void handleStatusChanged(TableRuntime tableRuntime, OptimizingStatus originalStatus) {
-    if (originalStatus != null && originalStatus.equals(OptimizingStatus.COMMITTING) &&
-        tableRuntime.getOptimizingStatus().equals(OptimizingStatus.IDLE)) {
-      tryEvaluatingPendingInput(tableRuntime, loadTable(tableRuntime));
-    }
-  }
-
   private void tryEvaluatingPendingInput(TableRuntime tableRuntime, ArcticTable table) {
     if (tableRuntime.isOptimizingEnabled() && !tableRuntime.getOptimizingStatus().isProcessing()) {
       OptimizingEvaluator evaluator = new OptimizingEvaluator(tableRuntime, table);

--- a/ams/server/src/main/java/com/netease/arctic/server/table/executor/TableRuntimeRefreshExecutor.java
+++ b/ams/server/src/main/java/com/netease/arctic/server/table/executor/TableRuntimeRefreshExecutor.java
@@ -18,7 +18,6 @@
 
 package com.netease.arctic.server.table.executor;
 
-import com.netease.arctic.server.optimizing.OptimizingStatus;
 import com.netease.arctic.server.optimizing.plan.OptimizingEvaluator;
 import com.netease.arctic.server.table.TableManager;
 import com.netease.arctic.server.table.TableRuntime;

